### DISCUSSION TITLE
Including Data from DB Query in Layout in all Pages

### DISCUSSION THREAD

**Discussion by CynthiaBlue** [quasarframework/quasar#12528] [CLI - Electron mode]
I use an IndexedDB database to store data, and manipulate it with Dexie.  I have a couple of database reads that I do quite often on various vue pages, and I would love to not have to duplicate the query on every page.

Is it possible to have the query be in the MainLayout.vue file (which I still use from the initial Quasar installation) and have the values available in all the pages that use that Layout?  If so, how?  I've been reading bout Routes but I don't know if that is what I'm looking for.  

I'm loading the data into an array in MainLayout.vue and I don't know how (or if) I can access that array in the other .vue pages in my application.  This will be an electron app.

Thanks!
[https://pawtap.net/pawtapNew/](https://pawtap.net/pawtapNew/)

**Comment by MilosPaunovic** [comment]
Take a look at state management libraries for `Vue`, especially `Vuex` or `Pinia`.

- [https://vuex.vuejs.org/](https://vuex.vuejs.org/)
- [https://pinia.vuejs.org/](https://pinia.vuejs.org/)
